lib/Controller/LoginController.php

```diff
@@ -330,22 +355,24 @@ private function login($profile)
                         $systemgroup->addUser($user);
                     }
                 }
-            }
+            } else {
+                // We only need to manage certain special groups.
+                // This is a separate branch, since it must not touch other group memberships.
 
-            // Manage administrator role
-            if (array_key_exists('is_admin', $attr) && $adminAttr = $attr['is_admin']) {
-                $systemgroup = $this->groupManager->get('admin');
-                if (array_key_exists($adminAttr, $profile) && $profile[$adminAttr]) {
-                    $systemgroup->addUser($user);
-                } else {
-                    $systemgroup->removeUser($user);
+                // Manage administrator role
+                if ($is_admin !== null) {
+                    $systemgroup = $this->groupManager->get('admin');
+                    if ($is_admin) {
+                        $systemgroup->addUser($user);
+                    } else {
+                        $systemgroup->removeUser($user);
+                    }
                 }
-            }
 
-            // Add default group if present
-            if ($defaultGroup = $this->config->getSystemValue('oidc_login_default_group')) {
-                if ($systemgroup = $this->groupManager->get($defaultGroup)) {
-                    $systemgroup->addUser($user);
+                // Add default group if present
+                if ($defaultGroup) {
+                    // We checked that the group exists above.
+                    $this->groupManager->get($defaultGroup)->addUser($user);
```

Comment:
Does `groupManager` do something special to avoid data races (somebody else removing the group) or is this fine?